core/controllers/contributor_dashboard.py

```diff
@@ -700,3 +703,190 @@ def post(self):
+                    suggestion_services.get_all_translation_contribution_stats(
+                        user_id))
+                self.values = {
+                    'translation_contribution_stats': _get_complete_stats(
+                        stats, True)
+                }
+
+            if contribution_subtype == feconf.CONTRIBUTION_SUBTYPE_REVIEW:
+                stats = suggestion_services.get_all_translation_review_stats(
+                    user_id)
+                self.values = {
+                    'translation_review_stats': _get_complete_stats(
+                        stats, False)
+                }
+
+        if contribution_type == feconf.CONTRIBUTION_TYPE_QUESTION:
+            if contribution_subtype == feconf.CONTRIBUTION_SUBTYPE_SUBMISSION:
+                stats = suggestion_services.get_all_question_contribution_stats(
+                    user_id)
+                self.values = {
+                    'question_contribution_stats': _get_complete_stats(
+                        stats, False)
+                }
+
+            if contribution_subtype == feconf.CONTRIBUTION_SUBTYPE_REVIEW:
+                stats = suggestion_services.get_all_question_review_stats(
+                    user_id)
+                self.values = {
+                    'question_review_stats': _get_complete_stats(stats, False)
+                }
+
+        self.render_json(self.values)
+
+
+class ContributorAllStatsSummariesHandler(base.BaseHandler):
+    """Provides all data about submitter and reviewer statistics."""
+
+    GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+    URL_PATH_ARGS_SCHEMAS = {
+        'username': {
+            'schema': {
+                'type': 'basestring'
+            }
+        }
+    }
+    HANDLER_ARGS_SCHEMAS = {
+        'GET': {}
+    }
+
+    @acl_decorators.can_fetch_all_contributor_dashboard_stats
+    def get(self, username):
+        """Handles GET requests."""
+
+        user_id = user_services.get_user_id_from_username(username)
+
+        stats = suggestion_services.get_all_contributor_stats(user_id)
+        response = {}
+
+        if stats.translation_contribution_stats is not None:
+            response['translation_contribution_stats'] = _get_complete_stats(
+                stats.translation_contribution_stats, True)
+
+        if stats.translation_review_stats is not None:
+            response['translation_review_stats'] = _get_complete_stats(
+                stats.translation_review_stats, False)
+
+        if stats.question_contribution_stats is not None:
+            response['question_contribution_stats'] = _get_complete_stats(
+                stats.question_contribution_stats, False)
+
+        if stats.question_review_stats is not None:
+            response['question_review_stats'] = _get_complete_stats(
+                stats.question_review_stats, False)
+
+        self.render_json(response)
+
+
+def _get_complete_stats(
+    stats_data: List[Union[
```

Comment:
```suggestion
    backend_stats: List[Union[
```

"Data" doesn't add much value in the name since pretty much everything can be considered "data". Added "backend" as a hint that we're converting a backend stats object.

Reply:
Done